test/persistence-hooks.suite.js

```diff
@@ -402,6 +449,112 @@ module.exports = function(dataSource, should) {
           });
       });
 
+      if (dataSource.connector.findOrCreate) {
+        it('triggers `persist` hook on findOrCreate when found', function(done) {
+          TestModel.observe('persist', pushContextAndNext());
+
+          TestModel.findOrCreate(
+            { where: { name: existingInstance.name } },
+            { name: existingInstance.name },
+            function(err, record, created) {
+              if (err) return done(err);
+
+              record.id.should.eql(existingInstance.id);
+              observedContexts.should.eql(aTestModelCtx({
+                data: {
+                  id: getLastGeneratedUid(),
```

Comment:
If I understand this code correctly, then `getLastGeneratedUid()` is the same as `existingInstance.id`, right? I am proposing to  change the assertion here and assert against `existingInstance.id`, as that makes the test easier to understand, at least to me. The point is to use the same expected value for `record.id` and `context.data.id`.